src/components/policy/policy_regular/src/update_status_manager.cc

```diff
@@ -121,7 +121,12 @@ void UpdateStatusManager::OnNewApplicationAdded(const DeviceConsent consent) {
     return;
   }
   app_registered_from_non_consented_device_ = false;
-  ProcessEvent(kOnNewAppRegistered);
+  if (kOnResetRetrySequence == last_processed_event_) {
+    current_status_.reset(new UpToDateStatus());
```

Comment:
@jacobkeeler We reset status here, because of, if last update was finished with update_needed status (failed for any reason), need started for new one app with the same status. This way we avoided the transition from update_needed to update_needed state (According to your comment [comment ](https://github.com/smartdevicelink/sdl_core/pull/3208#discussion_r381505225))